viewing_party/party.py

```diff
@@ -1,23 +1,141 @@
-# ------------- WAVE 1 --------------------
-
 def create_movie(title, genre, rating):
-    pass
+    new_movie = {
+        "title": title, 
+        "genre": genre, 
+        "rating": rating
+    }
+    if not title or not genre or not rating:
+        new_movie = None
+    return new_movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, movie_to_watch):
+    for i in range(len(user_data["watchlist"])):
+        if movie_to_watch in user_data["watchlist"][i]["title"]:
+            del user_data["watchlist"][i]
+            user_data["watched"].append(movie_to_watch)
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
 
+    total_rating = 0
+    count = len(user_data["watched"])
+    if user_data["watched"]:
+        for i in range(count):
+            total_rating += user_data["watched"][i]["rating"]
+        return total_rating/count
+    else:
+        return 0
+
+def get_most_watched_genre(user_data):
+    rating = {}
+    if not user_data["watched"]:
+        return None
+    else:
+        for i in range(len(user_data["watched"])):
+            genre_name = user_data["watched"][i]["genre"]
+            if genre_name not in rating:
+                rating[genre_name] = 1
+            else: 
+                rating[genre_name] += 1
+            popular = [(value, key) for key, value in rating.items()]
+        return max(popular)[1]

```

Comment:
👍  A few thoughts:

- You made a guard clause here! This means the `else` clause isn't needed and we can unindent the main 'meat' of your logic. We don't always need an `else` clause when utilizing `ifi. We can interpret `if` statements as "if and only if this expression is true, THEN we'll do this".
- Nice work utilizing list compression to generate a list of tuples. However, did we need to store the `value` along with the key? Could we have created a list of only the `key` aka the genre names and found the most popular from that collection?
- Good work using `max()`

```python
def get_most_watched_genre(user_data):
    rating = {}
    if not user_data["watched"]:
        return None

    for i in range(len(user_data["watched"])):
        genre_name = user_data["watched"][i]["genre"]
        if genre_name not in rating:
            rating[genre_name] = 1
        else: 
            rating[genre_name] += 1
        popular = [(value, key) for key, value in rating.items()]
        return max(popular)[1]
```